R/xy-datamapping.R

```diff
@@ -7,9 +7,58 @@ XYDataMapping <- R6::R6Class(
   public = list(
     x = NULL,
     y = NULL,
-    initialize = function(x, y) {
+    plotPropertiesNames = NULL,
+    color = NULL,
+    size = NULL,
+    shape = NULL,
+    linetype = NULL,
+    data = NULL,
+
+    initialize = function(x, y,
+                              color = NULL,
+                              size = NULL,
+                              shape = NULL,
+                              linetype = NULL,
+                              data = NULL,
+                              metaData = NULL) {
       self$x <- x
       self$y <- y
+
+      self$plotPropertiesNames <- c("color", "size", "shape", "linetype")
+
+      self$color <- color
+      self$size <- size
+      self$shape <- shape
+      self$linetype <- linetype
+
+
+      if (!is.null(data)) {
+        self$data <- self$getMappedData(data, metaData)
+      }
+    },
+
+    getMapData = function(data, metaData = NULL) {

```

Comment:
I think we need to rename this method to clarify it's purpose.
It's a getMethod that has side effects on the local data member.

Methods with side effects, in general, have no return values (it can, but it's rate)

This would be more something like `createDataMapping` or something like that.